core/src/main/java/org/elasticsearch/common/TimeBasedUUIDGenerator.java

```diff
@@ -40,17 +41,20 @@
         assert SECURE_MUNGED_ADDRESS.length == 6;
     }
 
-    /** Puts the lower numberOfLongBytes from l into the array, starting index pos. */
-    private static void putLong(byte[] array, long l, int pos, int numberOfLongBytes) {
-        for (int i=0; i<numberOfLongBytes; ++i) {
-            array[pos+numberOfLongBytes-i-1] = (byte) (l >>> (i*8));
-        }
+    // protected for testing
+    protected long currentTimeMillis() {
+        return System.currentTimeMillis();
```

Comment:
yeah, maybe we should focus on the order of bytes here and rethink generation separately?